### PR TITLE
Fix skip generation

### PR DESCRIPTION
**Pull request by hsiehjackson**
# What does this PR do ?

Use generation if user set `write_predictions_to_file` to `True` or `dataset.metric.name!=loss` 

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py [hunk 1]
```diff
@@ -402,21 +402,24 @@ def inference_step(self, dataloader_iter, batch_idx, mode, dataloader_idx=0):
...
-
-        output = self.predict_step(batch, batch_idx, dataloader_idx)
-
-        inputs_text = [self.tokenizer.ids_to_text(c.tolist()) for c in batch['contexts']]
-        labels_text = [self.tokenizer.ids_to_text(a.tolist()) for a in batch['answers']]
-        preds_text = [
-            self.tokenizer.ids_to_text(t[l.item() :][: data_cfg.get('tokens_to_generate')])
-            for t, l in zip(output['token_ids'], batch['context_lengths'])
-        ]
+        if data_cfg.get("write_predictions_to_file", False):
```
Comment (arendu) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py:405]: what about if the user wants accuracy, this will crash right?

**Review by arendu — CHANGES_REQUESTED** [review]

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py (hunk 1 above)
Comment (hsiehjackson) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py:405]: So you mean we can check if user set `write_predictions_to_file=True` or user use other metrics except `loss` we should do generation?

Comment (hsiehjackson) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py]: I add checking `metric != loss`.

**Review by arendu — APPROVED** [review]
LGTM!

**Pull request merged**